pkg/virt-controller/watch/snapshot.go

```diff
@@ -637,6 +637,95 @@ func (ctrl *SnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Virt
 	return nil
 }
 
+func (ctrl *SnapshotController) updateVolumeSnapshotStatuses(vm *kubevirtv1.VirtualMachine) error {
+	log.Log.V(3).Infof("Update volume snapshot status for VM [%s/%s]", vm.Namespace, vm.Name)
+
+	vmCopy := vm.DeepCopy()
+	var statuses []kubevirtv1.VolumeSnapshotStatus
+	for _, volume := range vmCopy.Spec.Template.Spec.Volumes {
+		log.Log.V(3).Infof("Update volume snapshot status for volume [%s]", volume.Name)
+		status := ctrl.getVolumeSnapshotStatus(vmCopy, &volume)
+		statuses = append(statuses, status)
+	}
+
+	vmCopy.Status.VolumeSnapshotStatuses = statuses
+	return ctrl.vmStatusUpdater.UpdateStatus(vmCopy)
+}
+
+func (ctrl *SnapshotController) getVolumeSnapshotStatus(vm *kubevirtv1.VirtualMachine, volume *kubevirtv1.Volume) kubevirtv1.VolumeSnapshotStatus {
+	if volume == nil {
+		return kubevirtv1.VolumeSnapshotStatus{
+			Name:    volume.Name,
+			Enabled: false,
+			Reason:  fmt.Sprintf("Volume is nil [%s]", volume.Name),
+		}
+	}
+	sc, err := ctrl.getVolumeStorageClassForVolume(vm.Namespace, volume)
+	if err != nil {
+		return kubevirtv1.VolumeSnapshotStatus{Name: volume.Name, Enabled: false, Reason: err.Error()}
+	}
+	if sc == "" {
+		return kubevirtv1.VolumeSnapshotStatus{
+			Name:    volume.Name,
+			Enabled: false,
+			Reason:  fmt.Sprintf("No Volume Snapshot Storage Class found for volume [%s]", volume.Name),
+		}
+	}
+
+	return kubevirtv1.VolumeSnapshotStatus{Name: volume.Name, Enabled: true}
+}
+
+func (ctrl *SnapshotController) getVolumeStorageClassForVolume(namespace string, volume *kubevirtv1.Volume) (string, error) {
+
+	if volume.VolumeSource.PersistentVolumeClaim != nil {
+		pvcKey := cacheKeyFunc(namespace, volume.VolumeSource.PersistentVolumeClaim.ClaimName)
+		obj, exists, err := ctrl.pvcInformer.GetStore().GetByKey(pvcKey)
+		if err != nil {
+			return "", err
+		}
+
+		if !exists {
+			log.Log.V(3).Infof("PVC not in cache [%s]", pvcKey)
+			return "", nil
+		}
+		pvc := obj.(*corev1.PersistentVolumeClaim)
+		return ctrl.getVolumeSnapshotClass(*pvc.Spec.StorageClassName)
+	}
+	if volume.VolumeSource.Ephemeral != nil {
```

Comment:
hmmm, not currently snapshotting ephemeral volumes since they are read only.  But maybe we should?

I say leave out for now and add back if/when we snapshot ephemeral.

Reply:
Only now I've noticed that handling ephemeral is the same as handling PVCs. Adding back support for ephemeral will be a matter of adding a clause in line 680.